tesseract/tesseract_environment/src/core/environment.cpp

```diff
@@ -616,6 +808,22 @@ bool Environment::setActiveContinuousContactManager(const std::string& name)
   return true;
 }
 
+tesseract_collision::DiscreteContactManager::Ptr Environment::getDiscreteContactManager() const
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+  if (!discrete_manager_)
+    return nullptr;
+  return discrete_manager_->clone();
+}
+
+tesseract_collision::ContinuousContactManager::Ptr Environment::getContinuousContactManager() const
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+  if (!continuous_manager_)
+    return nullptr;
+  return continuous_manager_->clone();
```

Comment:
Soon to be a CloneCache::clone()?

Reply:
I believe that would be good thing to speed things up.